web/packages/teleport/src/User/UserContext.tsx

```diff
@@ -54,18 +65,34 @@ export function useUser(): UserContextValue {
 
 export function UserContextProvider(props: PropsWithChildren<unknown>) {
   const { attempt, run } = useAttempt('processing');
+  const { clusterId } = useStickyClusterId();
 
   const [preferences, setPreferences] = useState<UserPreferences>(
     makeDefaultUserPreferences()
   );
 
+  const [clusterPreferences, setClusterPreferences] =
+    useState<UserClusterPreferences>(makeDefaultUserClusterPreferences());
+
+  async function loadClusterPreferences() {
+    const storedPreferences = storage.getUserClusterPreferences();
+    try {
+      const preferences = await service.getUserClusterPreferences(clusterId);
+      storage.setUserClusterPreferences(preferences);
+      setClusterPreferences(preferences);
+    } catch (error) {
+      if (storedPreferences) {
+        setClusterPreferences(storedPreferences);
+      }
+    }
+  }
```

Comment:
I have some concerns about this mechanism:
1. In a situation where I have a slow connection (or just the request takes more time), and I quickly switch between clusters I can end up with `clusterId` from a cluster A, but preferences from a cluster B. That would happen because we don't cancel the requests and a race condition can happen.
2. In case of an error, we read the cluster preferences for the last cluster that was fetched successfully (and this can be any cluster).

To solve 1. I think we should cancel the request when we change the cluster.

As of storing the preferences, I don't see much value in keeping them in local storage, and reading them only in case of an error (but maybe I don't see something). 
IMO much more useful would be caching the preferences per cluster in a map in the JS state. Thanks to this, we could avoid sending a request every time when the cluster is changed.
When doing this, we could probably omit the suggestion about cancelling the requests, but then we would have to store attempts in that cache, which may be problematic I think.

Reply:
Admittedly, this mechanism is just following previous conventions for how the rest of our user preferences are stored. I supposed we could change this one entirely but I'll have to reach out to Ryan to see why he made it this way to make sure I'm not missing anything.